### PR TITLE
Refactor to use bytes instead of string, other stuff is style related.

### PR DESCRIPTION
**Pull request by deckarep**
- [x] all unit-tests are updated and run
- [x] refactors to hold []byte instead of string
- [x] style updates to code around locking and naked returns.
- [x] fixed and updated travis

### CODE REVIEW

File: cache.go [hunk 1]
```diff
@@ -7,78 +7,79 @@ import (
...
+	sync.RWMutex
 	ttl   time.Duration
 	items map[string]*Item
 }
 
+// NewCache is a helper to create instance of the Cache struct
+func New(duration time.Duration) *Cache {
+	cache := &Cache{
+		ttl:   duration,
+		items: map[string]*Item{},
```
Comment (gsquire) [inline cache.go]: Do you want to call `make()` here?

File: cache.go [hunk 2]
```diff
@@ -7,78 +7,79 @@ import (
...
+
 	item, exists := cache.items[key]
 	if !exists || item.expired() {
-		data = ""
-		found = false
-	} else {
-		item.touch(cache.ttl)
-		data = item.data
-		found = true
+		return nil, false
```
Comment (gsquire) [inline cache.go:43]: Just a thought...is it possible for an item to be expired but still in the cache? Would the `evict` function hold the map and make sure it was gone before anything else could access it?

File: cache.go [hunk 3]
```diff
@@ -7,78 +7,79 @@ import (
...
-	cache.mutex.Unlock()
-}
-
-func (cache *Cache) startCleanupTimer() {
+func (cache *Cache) startEvictionTimer() {
 	duration := cache.ttl
 	if duration < time.Second {
 		duration = time.Second
 	}
+
```
Comment (gsquire) [inline cache.go:64]: Do we need the parentheses around `func()`? Seems like an odd style.

**Review by gsquire — COMMENTED** [review]
Some thoughts more than anything. We should also exercise some concurrent tests.

File: cache.go (hunk 1 above)
Comment (deckarep) [reply cache.go]: Changing to `make()` since that seems to read better anyway.

File: cache.go (hunk 2 above)
Comment (deckarep) [reply cache.go:43]: I think we covered this in our chat.

**Comment by gsquire** [comment]
LGTM

**Pull request merged**